sermant-plugins/sermant-mq-grayscale/mq-grayscale-rocketmq-plugin/src/main/java/io/sermant/mq/grayscale/rocketmq/utils/RocketMqReflectUtils.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.mq.grayscale.rocketmq.utils;
+
+import io.sermant.core.utils.ReflectUtils;
+
+import org.apache.rocketmq.client.impl.MQClientAPIImpl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * 5.0.x/5.1.x path difference class reflect utils
+ *
+ * @author chengyouling
+ * @since 2024-12-19
+ */
+public class RocketMqReflectUtils {
+    private RocketMqReflectUtils() {
+    }
+
+    /**
+     * get groups
+     *
+     * @param groupList GroupList
+     * @return groupList
+     */
+    public static Set<String> getGroupList(Object groupList) {
+        Optional<Object> groupSetsOpt = ReflectUtils.invokeMethodWithNoneParameter(groupList, "getGroupList");
+        return groupSetsOpt.map(groupSets -> (Set<String>) groupSets).orElseGet(HashSet::new);
+    }
+
+    /**
+     * get brokerDatas
+     *
```

Comment:
method names like "getGroupList" only used in 5.1.x? besides,mange these method name as constant or enum

Reply:
all version has the method, because of GroupList path changed, occurs method is directly invoke exception. So using reflect to get groupList.